tools/test/examples/examples_lib.py

```diff
@@ -498,4 +516,61 @@ def fetch_output_image(output):
             image = lines[index][7:]
             if os.path.isfile(image):
                 return image
-    return False
\ No newline at end of file
+    return False
+
+
+def _perform_post_build_checks(example_name, checks, elf_dir_path):
+    """
+    Perform post build checks.
+
+    The function raises a PostBuildCheckFailureError exceptions if an error is
+    encountered.
+    """
+    if not checks:
+        raise PostBuildCheckFailureError("No post build check specified")
+
+    # Find the elf file
+    elf_file = None
+    for dirpath, _, filenames in os.walk(elf_dir_path):
+        for filename in filenames:
+            if filename.endswith(".elf"):
+                elf_file = os.path.join(dirpath, filename)
+
+    if not elf_file:
+        raise PostBuildCheckFailureError(
+            "Cannot find ELF file in {}".format(elf_dir_path)
+        )
+
+    for check in set(checks):
+        SUPPORTED_POST_BUILD_CHECKS[check](example_name, elf_file)
+
+
+# Post-build check functions should be listed below and must raise a
+# a PostBuildCheckFailureError in case of failure
+
+def _post_build_check_floating_point(example_name, elf_file):
+    """Check if there are floating points in the executable."""
+
+    if example_name in POST_BUILD_CHECK_FLOATS_EXCLUDED_EXAMPLES:
+        return
+
+    float_symbols = elf_float_checker.check_float_symbols(elf_file)
+
+    if float_symbols:
+        raise PostBuildCheckFailureError(
+            "Floating point symbols found in executable: {}".format(
+                float_symbols
+            )
+        )
+
+# Specify the example project that should not be checked for floating point because they will always fail
+POST_BUILD_CHECK_FLOATS_EXCLUDED_EXAMPLES = [
```

Comment:
I don't think it is good to put such example exception data in side python code, the will cause script very difficult to maintain. Currently I think in the `examples.json`  file is the best place for those kind of data

Reply:
Waiting for confirmation from @jamesbeyond is the example should just be removed and therefore lose this exclusion code in the script.